src/traces/scattergl/convert.js

```diff
@@ -279,7 +279,8 @@ proto.updateFast = function(options) {
         yy = y[i];
 
         // check for isNaN is faster but doesn't skip over nulls
-        if(!isNumeric(xx) || !isNumeric(yy)) continue;
+        if(!isNumeric(yy)) continue;
+        if(!isNumeric(xx) && !(xx instanceof Date)) continue;
```

Comment:
[`Lib.isDateTime`](https://github.com/plotly/plotly.js/blob/f2d625c1de4b2938e2dada787f6c8fbe4904f55b/src/lib/dates.js#L107) is what you need here.

I'm afraid it will be significantly slower than `xx instanceof Date` though.